File: 10_Modelisation_base_de_données/exos/exo-redis-produit/class/Product.js
class Product {
    id;
    name;
    stock;
    price;
    static count = 0;

    constructor(name,stock,price){
        Product.count++;
        this.id = Product.count;
        this.name = name;
        this.stock = stock;
        this.price = price;
    };

    set name (name){
        this.name = name;
    }

    get name () {
        return this.name;
    }

    set stock(stock) {
        this.stock  = stock;
    }

    get stock() {
        return this.stock;
    }
    set price(price){
        this.price = price;
    }
    get price(){
        return this.price;
    }

}

module.exports = Product;

File: 18_Java/05_JPA/Exercice/TP_Product/src/main/java/Entity/Product.java
package Entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Data
public abstract class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    protected int id;
    protected String name;
    protected double  price;

    @Override
    public String toString() {
        return this.getClass().getSimpleName()+"(" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", price=" + price;
    }
}
